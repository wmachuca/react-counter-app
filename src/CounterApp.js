import React, { Fragment, useState } from 'react';
import PropTypes from 'prop-types';

const CounterApp = ({ value }) => {

    const [counter, setCounter] = useState(value);

    const handleAdd = () => {
        setCounter((c) => c + 1);
    }
    const handleReset = () => {
        setCounter(() => value);
    }
    const handleSubstract = () => {
        setCounter((c) => c - 1);
    }

    return (
        <Fragment>
            <h1>CounterApp</h1>
            <h2> {counter} </h2>
            <button id="btnAdd" onClick={handleAdd}>+1</button>
            <button id="btnReset" onClick={handleReset}>Reset</button>
            <button id="btnSubstract"onClick={handleSubstract}>-1</button>
        </Fragment>
    );
}

CounterApp.propTypes = {
    value: PropTypes.number
}

CounterApp.defaultProps = {
    value: 1
}

export default CounterApp;