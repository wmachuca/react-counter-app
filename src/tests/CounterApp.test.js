import CounterApp from '../CounterApp';
import { shallow } from 'enzyme';

describe('Prueba de CounterApp.js', () => {

    let wrapper = shallow(<CounterApp />);

    beforeEach(() => {
        wrapper = shallow(<CounterApp />);
    });

    test('Debe mostrar <CounterApp /> correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('Debe mostrar 100 en el valor por defecto', () => {
        const defaultValue = 100;
        const wrapper = shallow(<CounterApp value={defaultValue} />);
        const value = wrapper.find('h2').text().trim();
        expect(value).toBe(defaultValue.toString());
    })

    test('Debe incrementar el contador con el boton +1', () => {
        const initialValue = wrapper.find('h2').text().trim();
        wrapper.find('#btnAdd').simulate('click');
        const currentValue = wrapper.find('h2').text().trim();
        expect(currentValue).toBe((parseInt(initialValue) + 1).toString());
    })

    test('Debe decrementar el contador con el boton -1', () => {
        const initialValue = wrapper.find('h2').text().trim();
        wrapper.find('#btnSubstract').simulate('click');
        const currentValue = wrapper.find('h2').text().trim();
        expect(currentValue).toBe((parseInt(initialValue) - 1).toString());
    })

    test('Debe colocar el valor por defecto con el boton Reset', () => {
        const defaultValue = 100;
        const wrapper = shallow(<CounterApp value={defaultValue} />);
        wrapper.find('#btnAdd').simulate('click');
        wrapper.find('#btnReset').simulate('click');
        const currentValue = wrapper.find('h2').text().trim();
        expect(currentValue).toBe(defaultValue.toString());
    })

})
