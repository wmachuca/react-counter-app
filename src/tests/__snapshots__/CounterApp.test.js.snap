// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de CounterApp.js Debe mostrar <CounterApp /> correctamente 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
     
    1
     
  </h2>
  <button
    id="btnAdd"
    onClick={[Function]}
  >
    +1
  </button>
  <button
    id="btnReset"
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    id="btnSubstract"
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
